src/vm-vcpu/src/vcpu/mod.rs

```diff
@@ -187,6 +316,67 @@ impl KvmVcpu {
         Ok(vcpu)
     }
 
+    #[cfg(target_arch = "x86_64")]
+    fn set_state(&mut self, state: VcpuState) -> Result<()> {
```

Comment:
I wonder if it's safer to have this method consume self instead of mutating it in place. At the moment it's only used in `from_state()` below where the usage is def safe because a new `KvmVcpu` object is created locally and either returned being correct or dropped in case of an error - a malformed object is never returned.

However, the method in its current form does not prevent setting an existing object to incorrect state and keep using it if the client code is not careful.

Reply:
Since this is an internal function, I think I would rather add a comment instead and say that the errors coming from this function must be immediately return because otherwise you end up in an undefined state. Does that sound right to you?